src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.SpeculativeMemberSemanticModel.cs

```diff
@@ -19,17 +20,13 @@ private sealed class SpeculativeMemberSemanticModel : MemberSemanticModel
             /// Creates a speculative SemanticModel for a TypeSyntax node at a position within an existing MemberSemanticModel.
             /// </summary>
             public SpeculativeMemberSemanticModel(SyntaxTreeSemanticModel parentSemanticModel, Symbol owner, TypeSyntax root, Binder rootBinder, int position)
-                : base(root, owner, rootBinder, containingSemanticModelOpt: null, parentSemanticModelOpt: parentSemanticModel, speculatedPosition: position)
+                : base(root, owner, rootBinder, containingSemanticModelOpt: null, parentSemanticModelOpt: parentSemanticModel, snapshotManagerOpt: null, speculatedPosition: position)
             {
             }
 
-            protected override BoundNode RewriteNullableBoundNodesWithSnapshots(BoundNode boundRoot, Binder binder, DiagnosticBag diagnostics, out NullableWalker.SnapshotManager snapshotManager)
+            protected override BoundNode RewriteNullableBoundNodesWithSnapshots(BoundNode boundRoot, Binder binder, DiagnosticBag diagnostics, bool createSnapshots, out NullableWalker.SnapshotManager snapshotManager)
             {
-                // https://github.com/dotnet/roslyn/issues/35037: Speculative models are going to have to do something more advanced
-                // here. They will need to run nullable analysis up to the point that is being speculated on, and
-                // then take that state and run analysis on the statement or expression being speculated on.
-                // Currently, it will return incorrect info because it's just running analysis on the speculated
-                // part.
+                Debug.Assert(boundRoot.Syntax is TypeSyntax);
```

Comment:
>Debug.Assert(boundRoot.Syntax is TypeSyntax); [](start = 16, length = 45)

Do we even need to do any nullable analysis in this case?

Reply:
Currently, yes. The `TypeSyntax` will be turned into a `BoundTypeExpression`. We could make the `BoundTypeExpression` pull its nullability info from the underlying syntax except for `var`, but we're not currently doing that.

---
In reply to: [295528113](https://github.com/dotnet/roslyn/pull/36563#discussion_r295528113) [](ancestors = 295528113)